RedHat-UX/red-hat-design-system
# tools(proxy): adding Spandx configuration to proxy RHDC while developing

## What I did

1. Adding [spandx](https://github.com/redhataccess/spandx) config for RHDC.  This will allow us to proxy RHDC and other properties while developing/testing our components.
2. Added `/proxy/index.html` that is injected into the bottom of the page via the proxy.  This means that all content contained in this file will be present in the source html on page load.
3. Any request to a file at `/elements/` will be proxied to `web-dev-server`


## Testing Instructions

1.  Start the dev server `npm run dev`. Ensure it's running on port `8000`
2. In a separate window start the proxy `npm run proxy`
3. Open the url from the proxy server. (typically http://localhost:8001/en)
4. You should see the RHDC homepage.
5. Scroll all the way to the bottom, you should see the new `rh-footer`

## ToDos

- [ ] Decide how we should allow devs to edit the proxy / spandx settings.
- [x] Decide where we should allow devs to edit the contents of `proxy/index.html`
- [x] Get Spandx to proxy with `TS` files
- [ ] Browsersync can't find client-side js file, throws error in server console.  Occasionally works though 🤔
